paasta_tools/paasta_maintenance.py

```diff
@@ -472,6 +474,29 @@ def is_safe_to_kill(hostname):
     return is_host_drained(hostname) or is_host_past_maintenance_start(hostname)
 
 
+def is_hostname_local(hostname):
+    return hostname == 'localhost' or hostname == getfqdn() or hostname == gethostname()
+
+
+def is_safe_to_drain(hostname):
+    """Checks if a host has healthy tasks running locally that have low
+    replication in other places
+    :param hostname: hostname to check
+    :returns: True or False
+    """
+    if not is_hostname_local(hostname):
+        print (
+            "Due to the way is_safe_to_drain is implemented, it can only work "
+            "on localhost."
+        )
+        return False
+    return are_local_tasks_in_danger()
+
+
+def are_local_tasks_in_danger():
+    return True
```

Comment:
The name of this function implies that all local tasks are assumed to be in danger until you create an implementation.

I'm assuming that you would not e.g. run puppet if local tasks are in danger, so as this is written, you'd be preventing anything with a pre-flight check from running.